src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/TempDataDictionary.cs

```diff
@@ -101,6 +101,13 @@ public ICollection<object> Values
         /// <inheritdoc />
         public void Keep()
         {
+            // if the data is not loaded, we can assume none of it has been read
+            // and so silently return.
+            if (!_loaded)
```

Comment:
This check is needed to prevent from throwing exception (ex: 'Session feature not enabled') in cases where use doesn't even use TempData...example: you could be redirecting to another action (this was caught in some failing tests).
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/SaveTempDataFilter.cs#L45